lib/pwsh/util.rb

```diff
@@ -12,7 +12,9 @@ module Util
     def on_windows?
       # Ruby only sets File::ALT_SEPARATOR on Windows and the Ruby standard
       # library uses that to test what platform it's on.
-      !!File::ALT_SEPARATOR
+      require 'rbconfig'
+      is_windows = (RbConfig::CONFIG['host_os'] =~ /mswin|mingw|cygwin/)
```

Comment:
```suggestion
      is_windows = RbConfig.windows_platform?
```
Just a small suggestion for readability. https://rubydoc.info/gems/library/RbConfig.windows_platform%3F

Reply:
@jordanbreen28 Thanks for the feedback,

There are some drawbacks with the proposed usage of method `RbConfig.windows_platform?`:
- The method `RbConfig.windows_platform?` isn't shipped with Ruby core. It requires the gem `Library` `0.1.0`. 
  - Adding a gem to a module that isn't shipped with Puppet adds a burden to end-users to distribute the gem themselves.   
  - The referenced gem [library](https://rubygems.org/gems/library) has a single release v0.1.0 from 2012. This appears to be abandoned.
  - The more recently released gem [facets](https://www.rubydoc.info/github/rubyworks/facets) contains an equivalent module method `RbConfig.windows?`. The method compares less options in the pattern however has the same intent.

Instead of adding a gem dependency, we could add code comments for readability:
- We have done this a few times already in Pwshlib. 
  - See `Pwsh::Util.snake_case`. The comments indicate we copied the implementation from `facets`. 
https://github.com/chambersmp/ruby-pwsh/blob/56a4b81ca069a2de7eaff812fe1a607c334c5b69/lib/pwsh/util.rb#L48-L63
- This would avoid daisy chaining the Utils method `Pwsh::Util.on_windows?` to another method in a different gem. 
- The implementation would emulate that of the `library` or `facets` equivalents and does not require complex gem handling which is a win for consumers.

Reply:
Ah! I hadn't noticed that it. That's great thanks for clearing that up. I mean this all looks good on my end now